taichi/com_mgr/kernel_compilation_manager.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <ctime>
+#include <string>
+#include <memory>
+#include <unordered_map>
+
+#include "taichi/util/offline_cache.h"
+#include "taichi/codegen/kernel_compiler.h"
+#include "taichi/codegen/compiled_kernel_data.h"
+
+namespace taichi::lang {
+
+struct CacheData {
+  enum CacheMode { MemCache, MemAndDiskCache };
```

Comment:
nit: docstring to explain each enum?

Reply:
does `MemAndDiskCache` mean we cache to memory and disk simultaneously ?

Reply:
Yes. And FYI: after dumping the *in-memory cached data* to disk, they become *offline cached data*. So I think that we don't need to provide `DiskCache`.

Reply:
Yea I wonder in what scenario do we want to dump to disk earlier than taichi program exit? ;)

Reply:
Dump logic:
```
for k in cached_kernels:
  if k.cache_mode == `MemAndDiskCache`:
    dump the k
```